pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -124,6 +124,18 @@ func newEndpoint(e *model.NetworkEndpoint) (*endpoint.LbEndpoint, error) {
 		},
 	}
 
+	if e.UID != "" {
+		ep.Metadata = &core.Metadata{
+			FilterMetadata: map[string]*types.Struct{
+				"mixer": &types.Struct{
+					Fields: map[string]*types.Value{
+						"uid": &types.Value{Kind: &types.Value_StringValue{StringValue: e.UID}},
```

Comment:
Are you sure this works? Because the mixer filter does not know the specific endpoint envoy will talk to at the time of processing the HTTP request/TCP connection. So, I don't see how this info can be used by the filter on the outbound path.

Reply:
I'm 100% sure it works. It's only utilized by telemetry, post-routing.